packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -695,6 +702,24 @@ mixin WidgetsBinding on BindingBase, SchedulerBinding, GestureBinding, RendererB
       debugBuildingDirtyElements = true;
       return true;
     }());
+
+    if (_needToReportFirstFrame && _reportFirstFrame) {
+      assert(!_firstFrameCompleter.isCompleted);
+      // TODO(liyuqian): use a broadcast stream approach
+      final TimingsCallback oldCallback = WidgetsBinding.instance.window.onReportTimings;
+      WidgetsBinding.instance.window.onReportTimings = (List<FrameTiming> timings) {
+        if (!kReleaseMode) {
+          developer.Timeline.instantSync('Rasterized first useful frame');
+          developer.postEvent('Flutter.FirstFrame', <String, dynamic>{});
```

Comment:
Is this happening after the frame has been presented or rasterized?  The language of this PR is talking about rasterization but what I found out is that what I was really interested in is not when the frame was rasterized but when it was presented, which is a subtle difference.

Reply:
I'll revise the language to make it clear that:
1. We're only waiting for rasterization
2. Usually, the time that a frame is rasterized is very close to the time that it gets presented. Specifically, rasterization is the last expensive phase of a frame that's still in Flutter's control.